File: DataStructure/2.QueueAndDeque/Group2/hyunjin/10828 스택.py
import sys

stack = []
n = int(input())  # 명령의 수를 입력받음

for i in range(n):
    s = sys.stdin.readline().split()

    if s[0] == 'push':
        stack.append(s[1])  # 'append'를 올바르게 사용

    elif s[0] == 'pop':
        if len(stack) > 0:
            print(stack.pop())
        else:
            print(-1)

    elif s[0] == 'size':
        print(len(stack))

    elif s[0] == 'empty':
        if len(stack) == 0:
            print(1)
        else:
            print(0)

    elif s[0] == 'top':  # 'top' 명령이 누락된 부분 추가
        if len(stack) > 0:
            print(stack[-1])
        else:
            print(-1)

File: DataStructure/1.Stack/Group2/sungchan/10828 - 스택.cpp
#include <bits/stdc++.h>
using namespace std;
int main(void){
  ios::sync_with_stdio(0);
  cin.tie(0);
  stack<int> arr;
  int a = 0;
  cin >> a;
  while(a--){
      string name;
      cin >> name;
      if(name == "push"){
        int input = 0;
        cin >> input; 
        arr.push(input);
      }
      else if(name == "pop"){
        if(arr.empty())
          cout << -1 << '\n';
        else{
          cout << arr.top() << '\n';
          arr.pop();
        }
      }
      else if(name == "top"){
        if(arr.empty())
          cout << -1 << '\n';
        else 
          cout << arr.top() << '\n';
      }
      else if(name == "size")
          cout << arr.size() << '\n';
      else if(name == "empty")
        cout << arr.empty() << '\n';
    }
  return 0;